src/averageAges.js

```diff
@@ -36,8 +54,22 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const haveChildren = (all, name) =>
+    all.some(person => person.mother === name);
+
+  const WOMEN = withChildren
+    ? people.filter(person =>
+      (person.sex === 'f' && haveChildren(people, person.name)))
+    : people.filter(person => person.sex === 'f');
+
+  const WOMEN_TOTAL_AGE = WOMEN
+    .reduce((v, person) => v + (person.died - person.born), 0);
+
+  const WOMEN_AVERAGE_AGE = WOMEN_TOTAL_AGE / WOMEN.length;
+
+  return Number(WOMEN_AVERAGE_AGE.toFixed(2));
```

Comment:
typically we use uppercased variables for hardcoded values, these are calculated